### ISSUE TITLE
Add scoped route key name

### ISSUE THREAD

**Issue by bkuhlmann** [hanami/hanami#1297]
## Overview

Hello. :wave: I ran into a situation where I needed to *scope* my routes and noticed that the name you use for a scope doesn't resolve to the associated action properly.

## Steps to Recreate

Consider the following route:

``` ruby
module Demo
  class Routes < Hanami::Routes
    slice :api, at: "/api" do
      scope :chat do
        post "/slack", to: "slack.create", as: :slack_create
      end
    end
  end
end
```

...and here's the directory structure:

``` 
demo/slices/api/actions/chat/slack/create.rb
```

In order to resolve the above you have use `chat.slack.create` instead of `slack.create`. Shouldn't the scope name be included by default? I'm not sure if this is a bug or a feature request so opted to log this as the latter.

## Environment

- **Ruby**: ruby 3.2.1 (2023-02-08 revision 31819e82c8) +YJIT [arm64-darwin22.3.0]
- **Hanami**: 2.0.3

**Comment by solnic** [comment]
I don't think it should. You may want to use whatever object as the handler. Setting the namespace implicitly would make it impossible.

What I would suggest instead would be setting `to:` by default by inferring its value from the context and the path.

**Comment by bkuhlmann** [comment]
Actually, yes, that would be a nice quality of life improvement. Expanding upon this, it would be nice to take this boilerplate code:

``` ruby
slice :api, at: "/api" do
  get "/tasks", to: "tasks.index", as: :tasks
  get "/tasks/:id", to: "tasks.show", as: :task
  get "/tasks/new", to: "tasks.new", as: :task_new
  get "/tasks/:id/edit", to: "tasks.edit", as: :task_edit
  post "/tasks", to: "tasks.create", as: :task_create
  patch "/tasks/:id", to: "tasks.patch", as: :task_patch
  put "/tasks/:id", to: "tasks.update", as: :task_update
  delete "/tasks/:id", to: "tasks.delete", as: :task_delete

  scope :chat do
    post "/slack", to: "chat.slack.create", as: :slack_create
  end
end
```

...and turn it into this:

``` ruby
slice :api do
  get "/tasks"
  get "/tasks/:id"
  get "/tasks/new"
  get "/tasks/:id/edit"
  post "/tasks"
  patch "/tasks/:id"
  put "/tasks/:id"
  delete "/tasks/:id"

  scope :chat do
    post "/slack"
  end
end
```

The last code snippet would be identical in behavior to the first snippet but without all of the typing! You could then override the _inferred_ defaults -- per your suggestion -- by customizing `:at`, `:to`, and `:as` as you see fit but if you just wanted to roll with the inferred defaults you'd safe yourself a lot of time.

Happy to rewrite this issue to reflect the above if you think this would be a welcomed change?

**Comment by bkuhlmann** [comment]
**Peter**: I opted to rename and reword this issue based on our discussion since the resolution would have more impact to the community at large.

**Comment by 4D-Coder** [comment]
Hi! I'm just introducing myself to Hanami and would like to try my hand at this issue!
